### DISCUSSION TITLE
Modify API of canny to support per batch thresholds

### DISCUSSION THREAD

**Discussion by kunaltyagi** [kornia/kornia#2692] [General]
Right now, canny has a single threshold for all images in a batch. I want to add support for a threshold per image, but:
* It would require extending the API (and checks)
* It might cause perf regressions (needs testing basically)

I'd like to know if the API change would be acceptable (from just float to float or torch.Tensor). If yes, I would also like some guidance for the benchmarking

**Comment by edgarriba** [comment]
@kunaltyagi i think that would be a great addition. Also an opportunity the revisit the implementation. /cc @ducha-aiki

**Comment by johnnv1** [comment]
The benchmark can be done with https://pytorch.org/tutorials/recipes/recipes/benchmark.html here https://github.com/kornia/kornia-benchmark we have some examples using it, we can also discuss further on the pr
